roxie/ccd/ccdserver.cpp

```diff
@@ -11685,7 +11685,8 @@ class CRoxieServerDiskWriteActivity : public CRoxieServerInternalSinkActivity, i
         diskmeta.set(helper.queryDiskRecordSize()->querySerializedDiskMeta());
         if (grouped)
             diskmeta.setown(createDeltaRecordSize(diskmeta, +1));
-        blockcompressed = (((helper.getFlags() & TDWnewcompress) != 0) || (((helper.getFlags() & TDXcompress) != 0) && (diskmeta->getFixedSize() >= MIN_ROWCOMPRESS_RECSIZE))); //always use new compression
+        size32_t fixedSize = diskmeta->getFixedSize();
+        blockcompressed = (((helper.getFlags() & TDWnewcompress) != 0) || (((helper.getFlags() & TDXcompress) != 0) && ((0 == fixedSize) || (fixedSize >= MIN_ROWCOMPRESS_RECSIZE)))); //always use new compression
```

Comment:
@richardkchapman - I suspect this isn't an issue in Roxie.

It was an issue in similar code in hthor, that cause it not to compress spill files if they were variable length.
But there will be no spill activities in Roxie right?

Reply:
Roxie does not spill